lib/OpenQA/WebAPI/Controller/Running.pm

```diff
@@ -118,7 +112,7 @@ sub streamtext {
     # Check for new lines from the logfile using recurring timer
     # Setup utility function to close the connection if something goes wrong
     my $timer_id;
-    my $close_connection = sub {
+    my $close_connection = sub ($self) {
```

Comment:
I think the signature should be empty because below it is called like this without any arguments:
`$close_connection->()`

Reply:
done